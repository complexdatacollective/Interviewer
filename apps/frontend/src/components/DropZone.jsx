import React from 'react';
import PropTypes from 'prop-types';

import { DropTarget } from '../behaviours/DragAndDrop';

/**
  * Renders a container onto which a `draggable` can be dropped.
  */
const DropZone = ({ children }) => (
  <div className="drop-zone">
    {children}
  </div>
);

DropZone.propTypes = {
  children: PropTypes.any,
};

DropZone.defaultProps = {
  children: null,
};

export default DropTarget(DropZone);
