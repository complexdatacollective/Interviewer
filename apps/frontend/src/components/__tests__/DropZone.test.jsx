/* eslint-env jest */

import React from 'react';
import { createStore } from 'redux';
import { shallow } from 'enzyme';
import DropZone from '../DropZone';

const mockStore = createStore(() => ({ droppable: { activeZones: [] } }));

describe('DropZone component', () => {
  it('renders ok', () => {
    const component = shallow(<DropZone id="foo" store={mockStore} />);

    expect(component).toMatchSnapshot();
  });
});
