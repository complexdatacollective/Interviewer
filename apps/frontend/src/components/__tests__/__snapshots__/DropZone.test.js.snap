// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DropZone component renders ok 1`] = `
<DropZone
  id="foo"
  onDrag={[Function]}
  onDragEnd={[Function]}
  onDrop={[Function]}
  store={
    Object {
      "@@observable": [Function],
      "dispatch": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
    }
  }
/>
`;
